bothan-coinbase/src/api/websocket.rs

```diff
@@ -24,21 +24,23 @@ impl WebSocketConnector {
     }
 
     /// Connects to the WebSocket and returns a `CoinbaseWebSocketConnection`.
-    pub async fn connect(&self) -> Result<WebSocketConnection, ConnectionError> {
-        let (wss, resp) = connect_async(self.url.clone()).await?;
-
-        let status = resp.status();
-        if StatusCode::is_server_error(&status) || StatusCode::is_client_error(&status) {
-            warn!("failed to connect with response code {}", resp.status());
-            return Err(ConnectionError::UnsuccessfulWebSocketResponse(
-                resp.status(),
-            ));
-        }
+    pub async fn connect(&self) -> Result<WebSocketConnection, tungstenite::Error> {
+        let (wss, _) = connect_async(&self.url).await?;
 
         Ok(WebSocketConnection::new(wss))
     }
 }
 
+#[async_trait::async_trait]
+impl AssetInfoProviderConnector for WebSocketConnector {
+    type Provider = WebSocketConnection;
```

Comment:
Do we use this?

Reply:
we use it in the worker function refer to start_polling() in worker/websocket